multidim-interop/rust/src/lib.rs

```diff
@@ -1,13 +1,44 @@
-use std::env;
 use std::time::Duration;
 
 use anyhow::{Context, Result};
 use env_logger::Env;
 use log::info;
 use redis::{AsyncCommands, Client as Rclient};
+use strum::EnumString;
 
 const REDIS_TIMEOUT: usize = 10;
 
+/// Supported transports by rust-libp2p.
+#[derive(Clone, Debug, EnumString)]
+#[strum(serialize_all = "kebab-case")]
+pub enum Transport {
+    Tcp,
+    QuicV1,
+    Webrtc,
+    Ws,
+}
```

Comment:
With `clap` you could do something along the lines of:

``` rust
#[derive(clap::ValueEnum, Clone)]
pub enum Transport {
    Tcp,
    QuicV1,
    Webrtc,
    Ws,
}

#[derive(Debug, Parser)]
#[clap(name = "rust-libp2p Interop Tests")]
struct Opts {
    /// The transport to listen on.
    #[clap(long)]
    transport: Transport,
}

fn main() {
    let opts = Opts::parse();
}
```

(Haven't actually tried the above.)

We are using `clap` in many places anyways, thus consistent across the ecosystem. It gives you nice parser errors. It gives us a `--help` text for free.

@jxs what do you think? I don't have a strong opinion on it. Feel free to ignore entirely or delay for another pull request.

Reply:
thanks for the input Max, and sorry I should have probably justified the decision. 
I actually started by following your suggestion with `clap`. But the thing is, your example assumes the variables are passed by command line options, but in our case are passed by environment variables. `clap` allows to address that with the `env` feature, but then `clap` expects the name of the variable to be uppercase, whereas in our case (transport) it's in lower case. 
I tried using [`rename_all`](https://docs.rs/clap/4.1.1/clap/_derive/index.html#arg-attributes) but it seems it only applies to the variable value. 

Then as I noticed in our case we just need to convert from `String`s to `Enum`s I resorted to `strum`.
If you prefer I can try to see if using the `clap` builder we can try to achieve the same.

Reply:
created a helper function `from_env` on `lib.rs` which basically made me feel that we are re-writing `clap` so I'll look after if `clap` can actually make this more elegant.

Reply:
Let's delay this :+1: